src/node_api.cc

```diff
@@ -710,7 +724,11 @@ napi_status napi_define_class(napi_env env,
                                     v8impl::FunctionCallbackWrapper::Invoke,
                                     cbdata,
                                     v8::Signature::New(isolate, tpl));
-      t->SetClassName(property_name);
+
+      // Class name has to be an identifier, though property names can be any
+      // property key. Enforce that class name is a string here.
+      RETURN_STATUS_IF_FALSE(property_name->IsString(), napi_string_expected);
+      t->SetClassName(property_name.As<v8::String>());
```

Comment:
Actually I don't think `SetClassName()` should be called at all here. The V8 doc for that API says:

> This is used for printing objects created with the function created from the FunctionTemplate as its constructor. 

What this was intended to do was set the function (class method) name. It's odd that `v8::FunctionTemplate` doesn't have a `SetName()` method, while `v8::Function` does.